### ISSUE TITLE
Once-only restriction on SendHeader and SetTrailer precludes their use in Unary interceptor

### ISSUE THREAD

**Issue by zellyn** [grpc/grpc-go#733]
https://godoc.org/google.golang.org/grpc#SendHeader and https://godoc.org/google.golang.org/grpc#SetTrailer both state that they may only be called once. How does one add headers or trailers in a server-side interceptor?

**Comment by iamqizhao** [comment]
Metadata API was designed without a plan to support interceptor at that time. Actually, I think SetTrailer works (i.e., it can be called multiple times before eventually returning from the interceptor; I need to verify though). For Header metadata, probably we can add another SetHeader API which stores metadata without sending (all metadata will be sent out when SendHeader is called.).

**Comment by NeoCN** [comment]
@iamqizhao any updates on this issue? In our code, we want to add one common header in server-side interceptor, but with the Once-only restriction, we can not do it!

**Comment by menghanl** [comment]
`SetTrailer` will work after #863. Will have another PR for `SendHeader`.
